pkg/asset/manifests/powervs/cluster.go

```diff
@@ -0,0 +1,238 @@
+package powervs
+
+import (
+	"fmt"
+	"reflect"
+
+	"github.com/sirupsen/logrus"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/utils/ptr"
+	capibm "sigs.k8s.io/cluster-api-provider-ibmcloud/api/v1beta2"
+
+	"github.com/openshift/installer/pkg/asset"
+	"github.com/openshift/installer/pkg/asset/installconfig"
+	"github.com/openshift/installer/pkg/asset/manifests/capiutils"
+	powervstypes "github.com/openshift/installer/pkg/types/powervs"
+)
+
+// GenerateClusterAssets generates the manifests for the cluster-api.
+func GenerateClusterAssets(installConfig *installconfig.InstallConfig, clusterID *installconfig.ClusterID, bucket string, object string) (*capiutils.GenerateClusterAssetsOutput, error) {
+	var (
+		manifests          []*asset.RuntimeFile
+		network            string
+		service            capibm.IBMPowerVSResourceReference
+		resourceGroupName  string
+		vpcName            string
+		vpcRegion          string
+		transitGatewayName string
+		cosName            string
+		cosRegion          string
+		imageName          string
+		bucketName         string
+		err                error
+		powerVSCluster     *capibm.IBMPowerVSCluster
+		powerVSImage       *capibm.IBMPowerVSImage
+	)
+
+	defer func() {
+		logrus.Debugf("GenerateClusterAssets: installConfig = %+v, clusterID = %v, bucket = %v, object = %v", installConfig, *clusterID, bucket, object)
+		logrus.Debugf("GenerateClusterAssets: ic.ObjectMeta = %+v", installConfig.Config.ObjectMeta.Name)
+		logrus.Debugf("GenerateClusterAssets: installConfig.Config.PowerVS = %+v", *installConfig.Config.PowerVS)
+		logrus.Debugf("GenerateClusterAssets: resourceGroupName = %v", resourceGroupName)
+		logrus.Debugf("GenerateClusterAssets: vpcName = %v", vpcName)
+		logrus.Debugf("GenerateClusterAssets: vpcRegion = %v", vpcRegion)
+		logrus.Debugf("GenerateClusterAssets: transitGatewayName = %v", transitGatewayName)
+		logrus.Debugf("GenerateClusterAssets: cosName = %v", cosName)
+		logrus.Debugf("GenerateClusterAssets: cosRegion = %v", cosRegion)
+		logrus.Debugf("GenerateClusterAssets: imageName = %v", imageName)
+		logrus.Debugf("GenerateClusterAssets: bucketName = %v", bucketName)
+		logrus.Debugf("GenerateClusterAssets: powerVSCluster.Spec.ControlPlaneEndpoint.Host = %v", powerVSCluster.Spec.ControlPlaneEndpoint.Host)
+	}()
+
+	manifests = []*asset.RuntimeFile{}
+
+	network = fmt.Sprintf("%s-network", clusterID.InfraID)
+
+	if installConfig.Config.PowerVS.ServiceInstanceGUID == "" {
+		serviceName := fmt.Sprintf("%s-iaas", clusterID.InfraID)
+
+		service = capibm.IBMPowerVSResourceReference{
+			Name: &serviceName,
+		}
+	} else {
+		service = capibm.IBMPowerVSResourceReference{
+			ID: &installConfig.Config.PowerVS.ServiceInstanceGUID,
+		}
+	}
+
+	if installConfig.Config.Platform.PowerVS.PowerVSResourceGroup != "" {
+		// @TODO is it a name or id?
+		resourceGroupName = installConfig.Config.Platform.PowerVS.PowerVSResourceGroup
+	}
```

Comment:
```suggestion

```

Reply:
The TODO means future work/investigation.  I am trying to get this PR in so I don't have to juggle code for a month.